samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -112,20 +113,30 @@ def prepare(params: dict) -> dict:
     try:
         # initialize terraform application
         LOG.info("Initializing Terraform application")
-        run(["terraform", "init", "-input=false"], check=True, capture_output=True, cwd=terraform_application_dir)
+        invoke_subprocess_with_loading_pattern(
+            command_args={
+                "args": ["terraform", "init", "-input=false"],
+                "cwd": terraform_application_dir,
+                "stdout": PIPE,
+            }
+        )
 
         # get json output of terraform plan
         LOG.info("Creating terraform plan and getting JSON output")
-
         with osutils.tempfile_platform_independent() as temp_file:
-            run(
-                # input false to avoid SAM CLI to stuck in case if the Terraform project expects input, and customer
-                # does not provide it.
-                ["terraform", "plan", "-out", temp_file.name, "-input=false"],
-                check=True,
-                capture_output=True,
-                cwd=terraform_application_dir,
+            process_output = invoke_subprocess_with_loading_pattern(
```

Comment:
are we loading the `.` to stderr by default? There are CI runners that will error out that moment they see something printed to stderr, we have to be conscious of that.  

What is our current pattern for the dots, which stream do they land on ?

Reply:
I was looking at other places in the codebase where we do something similar and we typically default to `stderr`. For example, we write loading dots when building Docker images https://github.com/aws/aws-sam-cli/blob/develop/samcli/local/docker/lambda_image.py#L236.  We can default it to `stdout` instead here if it makes more sense.

Reply:
I see, maybe its less invasive that I thought it to be, still something to be conscious about.

Reply:
we use the stdout to write the `local invoke` output. I think to make it easier for people to read the lambda output programatically

Reply:
Fair, good callout.